src/workers/atlas_worker.js

```diff
@@ -8,13 +8,47 @@ This Source Code Form is “Incompatible With Secondary Licenses”, as defined
 */
 
 import PeriodicWorker from './periodic_worker';
+import {getDefaultAddress} from '../utils/web3_tools';
+import {Role} from '../services/roles_repository';
 
 export default class AtlasWorker extends PeriodicWorker {
-  constructor(logger) {
+  constructor(web3, dataModelEngine, rolesRepository, challengesRepository, logger) {
     super(5000, logger);
+    this.web3 = web3;
+    this.dataModelEngine = dataModelEngine;
+    this.rolesRepository = rolesRepository;
+    this.challengesRepository = challengesRepository;
+    this.nextUnprocessedBlock = 0;
+  }
+
+  async beforeWorkLoop() {
+    this.logger.info('Atlas worker starting');
+    const role = await this.rolesRepository.onboardedRole(getDefaultAddress(this.web3));
+    if (!role.is(Role.ATLAS)) {
+      this.logger.error('Cannot start atlas worker until being onboarded as ATLAS');
+      throw new Error('Not onboarded as ATLAS');
+    }
+    this.nextUnprocessedBlock = await this.web3.eth.getBlockNumber();
+  }
+
+  async tryToResolve({sheltererId, bundleId, challengeId}) {
+    this.logger.info({message: `Trying to fetch the bundle`, sheltererId, bundleId, challengeId});
+    await this.dataModelEngine.downloadBundle(bundleId, sheltererId);
+    await this.challengesRepository.resolveChallenge(challengeId);
```

Comment:
We should actually check if the atlas is at the current moment able to resolve it (e.g. it wasn't resolved yet, the atlas is not a shelterer, etc). This is especially important as we expect this rules to be extended shortly. 

IMO: best solution would be to delegate this question to the smart contract. Also: we could check this before downloading, for the first time, and then a second time just before calling resolve.